src/circshiftedarray.jl

```diff
@@ -94,5 +100,5 @@ Returns amount by which `s` is shifted compared to `parent(s)`.
 """
 shifts(s::CircShiftedArray) = s.shifts
 
-checkbounds(::CircShiftedArray, I...) = nothing
-checkbounds(::Type{Bool}, ::CircShiftedArray, I...) = true
+# checkbounds(::CircShiftedArray, I...) = nothing
+# checkbounds(::Type{Bool}, ::CircShiftedArray, I...) = true
```

Comment:
```suggestion
```

Reply:
Hey,
I checked timings:
```julia
julia> using BenchmarkTools

julia> f(w, v) = begin
       for i in eachindex(w)
                  w[i] = i == 1 ? v[1] - v[end] : v[i] - v[i-1]
              end
           end
f (generic function with 1 method)

julia> g(w, v) = w .= v .- ShiftedArrays.circshift(v, 1);

julia> v, w = rand(1000), rand(1000);

julia> @btime g($w, $v); # is based on the modulo idea 
  1.425 μs (0 allocations: 0 bytes)

julia> @btime f($w, $v);
  652.564 ns (0 allocations: 0 bytes)
  ```
  
I then switched the implementation to a non-mod one where `shifts` is processed at the beginning:
```
julia> @btime g($w, $v);
  457.609 ns (0 allocations: 0 bytes)
```
Performance is therefore ~3x better.
Unfortunately `shifts` is not equal to the input anymore. Is that a problem? Semantically it is still the same.